### PR TITLE
Refactor gitsigns to use nullable options

### PR DESCRIPTION
**Pull request by traxys**
This adds a few helpers for enums to refactor gitsigns.

It changes all options to be nullable, significantly (by 80 lines) reducing the length of an init.lua that only enables gitsigns.

### CODE REVIEW

File: lib/helpers.nix [hunk 1]
```diff
@@ -86,8 +86,10 @@ rec {
     );
 
     mkInt = default: mkNullable lib.types.int (toString default);
-    mkBool = default: mkNullable lib.types.bool (toString default);
+    mkBool = default: mkNullable lib.types.bool (if default then "true" else "false");
```
Comment (pta2002) [inline lib/helpers.nix:89]: why this change?

Comment (pta2002) [reply lib/helpers.nix:89]: nvm, got it, `toString true/false` returns `1/0`

**Review by pta2002 — APPROVED** [review]

**Comment by pta2002** [comment]
thanks!

File: lib/helpers.nix (hunk 1 above)
Comment (traxys) [reply lib/helpers.nix:89]: Worse than that, it returns "1"/""
